### PR TITLE
proper toggling of error messages with ConfigurationValid condition

### PR DESCRIPTION
**Pull request by gabemontero**
Fixes https://github.com/openshift/cluster-samples-operator/issues/21

Looks good with live testing in addition to unit test

### CODE REVIEW

File: pkg/stub/handler_test.go [hunk 1]
```diff
@@ -128,6 +128,27 @@ func TestWithBadArch(t *testing.T) {
...
 
+func TestConfigurationValidCondition(t *testing.T) {
+	h, sr, event := setup()
+	err := h.Handle(nil, event)
+	validate(true, err, "", sr, []v1alpha1.SamplesResourceConditionType{v1alpha1.SamplesExist}, []corev1.ConditionStatus{corev1.ConditionTrue}, t)
+	_, sr, event = setup()
+	sr.Spec.InstallType = "rhel8"
+	sr.ResourceVersion = "2"
+	err = h.Handle(nil, event)
+	validate(false, err, "trying to change installtype, which is not allowed, but also specificed", sr, []v1alpha1.SamplesResourceConditionType{v1alpha1.ConfigurationValid}, []corev1.ConditionStatus{corev1.ConditionFalse}, t)
```
Comment (bparees) [inline pkg/stub/handler_test.go]: s/specificed/specified/

Comment (bparees) [reply pkg/stub/handler_test.go]: (hopefully that would have failed your test..)

**Comment by bparees** [comment]
nit, otherwise lgtm

File: pkg/stub/handler_test.go (hunk 1 above)
Comment (gabemontero) [reply pkg/stub/handler_test.go]: it was misspelled in the src code too ... adjusting in both places

**Pull request merged**